geth/jail/internal/loop/loop.go

```diff
@@ -65,19 +83,24 @@ func (l *Loop) VM() *vm.VM {
 // Add puts a task into the loop. This signals to the loop that this task is
 // doing something outside of the JavaScript environment, and that at some
 // point, it will become ready for finalising.
-func (l *Loop) Add(t Task) {
+func (l *Loop) Add(t Task) error {
 	l.lock.Lock()
+	defer l.lock.Unlock()
+	if !l.accepting {
+		t.Cancel()
```

Comment:
Cancelling task here seems not needed. Task should not be started at this point, so nothing to cancel.

Reply:
From the issue description in #415:
"2.  `func (l *Loop) Add(t Task)` should be func `(l *Loop) Add(t Task) error` + should cancel a task in case of an error (closed loop will yield an error),
3.  The same as (2) with `func (l *Loop) Ready(t Task)`,"

I can take it out, but it is there because of these requirements.

Reply:
Oh, I think that should be remived from task description.

The general idea here is thet `loop` implements JS VM event loop, and we have control over types of tasks we use (fetch, looptask, etc). I don't see a case where we can add a task that is already running (actually, it would cool if API of package doesn't allow than).